enterprise_subsidy/apps/api_client/enterprise.py

```diff
@@ -0,0 +1,62 @@
+"""
+Enterprise api client for the subsidy service.
+"""
+import logging
+
+import requests
+from django.conf import settings
+
+from enterprise_subsidy.apps.api_client.base_oauth import BaseOAuthClient
+
+logger = logging.getLogger(__name__)
+
+
+class EnterpriseApiClient(BaseOAuthClient):
+    """
+    API client for calls to the enterprise service.
+    """
+    api_base_url = settings.LMS_URL + '/enterprise/api/v1/'
+    enterprise_customer_endpoint = api_base_url + 'enterprise-customer/'
+    enterprise_learner_endpoint = api_base_url + 'enterprise-learner/'
+    pending_enterprise_learner_endpoint = api_base_url + 'pending-enterprise-learner/'
+    course_enrollments_revoke_endpoint = api_base_url + 'licensed-enterprise-course-enrollment/license_revoke/'
+
+    def get_enterprise_customer_data(self, enterprise_customer_uuid):
+        """
+        Gets the data for an EnterpriseCustomer with a given UUID.
+
+        Arguments:
+            enterprise_customer_uuid (UUID): UUID of the enterprise customer associated with an enterprise
+        Returns:
+            response (dict): JSON response data
+        """
+        endpoint = f'{self.enterprise_customer_endpoint}{str(enterprise_customer_uuid)}/'
+        try:
+            response = self.client.get(endpoint)
+            response.raise_for_status()
+            return response.json()
+        except requests.exceptions.HTTPError as exc:
+            logger.error(
+                f'Failed to fetch enterprise customer data for {enterprise_customer_uuid} because {response.text}',
+            )
+            raise exc
+
+    def bulk_enroll_enterprise_learners(self, enterprise_customer_uuid, options):
```

Comment:
a lot of the logic relies on the structure of `options`, maybe we keep this as a low-level method but provide other signatures with more explicit arguments?

Reply:
related, the current iteration also requires the "caller" to know a lot about the possible structure of `options`